### ISSUE TITLE
clj-oauth 1.5.2 on clojars prints private key to stdout

### ISSUE THREAD

**Issue by ebaxt** [mattrepl/clj-oauth#44]
Hey. After upgrading from 1.5.1 to 1.5.2 I noticed our logs were filing up with lines like:

```
Private key:  #object[org.bouncycastle.jce.provider.JCERSAPrivateCrtKey 0x41e6c9ea RSA Private CRT Key
...
```
I eventually tracked it down to line 84 in signature.clj in version 1.5.2 that is uploaded to clojars.

```clojure
(defmethod sign :rsa-sha1
  [c ^String base-string & [token-secret]]
  (java.security.Security/addProvider
    (org.bouncycastle.jce.provider.BouncyCastleProvider.))
  (let [private-key (-> (:secret c)
                        java.io.StringReader.
                        org.bouncycastle.openssl.PEMReader.
                        .readObject
                        .getPrivate)
        signer (doto (java.security.Signature/getInstance "SHA1withRSA" "BC")
                 (.initSign private-key (java.security.SecureRandom.))
                 (.update (.getBytes base-string)))
        raw-sig (.sign signer)]
    -->>> (println "Private key: " private-key)
    (String. (Base64/encodeBase64 raw-sig))))
```

**Comment by mattrepl** [comment]
That's horrifying, thank you for reporting. The HEAD version doesn't have this information leak, but apparently I relied solely on tests passing before pushing the release and didn't notice a contributor leaving a debug print in the code.

I will push a new version with the print removed as soon as I have Internet access on my dev machine.

**Comment by mattrepl** [comment]
Version 1.5.3 has been pushed to clojars. Thanks again.

**Comment by ebaxt** [comment]
Thanks for the quick fix, it'll try the 1.5.3 release then :)
